x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/RolloverStep.java

```diff
@@ -63,7 +68,16 @@ public void evaluateCondition(IndexMetaData indexMetaData, Listener listener) {
             rolloverRequest.addMaxIndexDocsCondition(maxDocs);
         }
         getClient().admin().indices().rolloverIndex(rolloverRequest,
-                ActionListener.wrap(response -> listener.onResponse(response.isRolledOver(), new EmptyInfo()), listener::onFailure));
+            ActionListener.wrap(response -> listener.onResponse(response.isRolledOver(), new EmptyInfo()), exception -> {
+                if (exception instanceof ResourceAlreadyExistsException) {
+                    // This can happen sometimes when this step is executed multiple times
+                    logger.debug("index [{}] cannot roll over because the next index already exists, skipping to next step",
```

Comment:
Can you add logging of the exception on TRACE level (so that in the future if this ever occurs for a different reason we still have access to the full stacktrace):

```java
if (logger.isTraceEnabled()) {
    logger.debug(() -> new ParameterizedMessage("{} index cannot roll over ...", indexMetaData.getIndex()), exception);
} else {
    logger.debug("{} index cannot roll over ...", indexMetaData.getIndex());
}
```

Reply:
Done and done.